node/core/core.go

```diff
@@ -0,0 +1,40 @@
+package core
+
+import (
+	"go.uber.org/fx"
+
+	"github.com/celestiaorg/celestia-node/core"
+	"github.com/celestiaorg/celestia-node/rpc"
+)
+
+// Config combines all configuration fields for Core subsystem.
+type Config struct {
+	EnableRemote bool
+	Remote       struct {
+		Protocol   string
+		RemoteAddr string
+	}
+	EmbeddedConfig *core.Config
+}
+
+// DefaultConfig returns default configuration for Core subsystem.
+func DefaultConfig() *Config {
+	return &Config{
+		EmbeddedConfig: core.DefaultConfig(),
+		EnableRemote:   false,
```

Comment:
I think right now, we should have this default be `true` as we haven't implemented celestia node <> celestia node communication yet, and even when we do, we will need people to run nodes with embedded core proc.

Reply:
I don't get why remote Core should be a default. EnableRemote true means Core is remote and not embedded. Do we really want remote to be a default?

Reply:
No ignore, this commetn

Reply:
No ignore this comment. I confused `EnableRemote` as enabling the embedded core node itself. The behaviour right now is that default, a core node will be embedded in the full node yes?

Reply:
No ignore this comment. I confused `EnableRemote` as enabling the embedded core node itself. The behaviour right now is that default, a core node will be embedded in the full node yes?

Reply:
Yes, by default the Core is embedded. EnableRemote enables remote mode.

Reply:
Okay as long as core is embedded by default.